nautobot_ssot/integrations/device42/utils/nautobot.py

```diff
@@ -362,3 +363,14 @@ def apply_vlans_to_port(adapter, device_name: str, mode: str, vlans: list, port:
                 tagged_vlans.append(tagged_vlan)
         port.tagged_vlans.set(tagged_vlans)
         port.validated_save()
+
+
+def ensure_contenttypes_on_location_type(location_type: LocationType):
+    """Ensure that the required ContentTypes are on the specified Building LocationType.
+
+    Args:
+        location_type (LocationType): The specified LocationType to use when importing Building Locations.
+    """
+    for obj_type in [Location, RackGroup, Rack, Device, VirtualChassis, Prefix, VLAN]:
+        location_type.content_types.add(ContentType.objects.get_for_model(obj_type))
+    location_type.save()
```

Comment:
Was there a reason the save was required here? If so, a comment would be helpful

Reply:
We need to ensure that the correct ContentType is on the specified LocationType as we assign these object types to Locations. But I'll add a note in the docstrings.

Reply:
Or wait, it was a save() because validated_save() wasn't available when it was a signal. It's been moved since. I'll corrected to validated_save().

Reply:
Adding objects to an m2m shouldn't require a save unless there are signals that you want to fire on the LocationType.save call

Reply:
Ah, well that's just habit to always save things. I can remove though.